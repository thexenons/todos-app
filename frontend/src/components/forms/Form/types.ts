import { PropsWithChildren } from "react";
import { FieldValues, SubmitHandler } from "react-hook-form";

export interface FormProps<FieldsType extends FieldValues>
  extends PropsWithChildren {
  onSubmit: SubmitHandler<FieldsType>;
}
