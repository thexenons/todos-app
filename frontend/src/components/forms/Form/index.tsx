import { FieldValues, FormProvider, useForm } from "react-hook-form";
import { FormProps } from "./types";

const Form = <FV extends FieldValues>({
  onSubmit,
  children,
}: FormProps<FV>) => {
  const form = useForm<FV>();

  return (
    <FormProvider<FV> {...form}>
      <form onSubmit={form.handleSubmit(onSubmit)}>{children}</form>
    </FormProvider>
  );
};

export default Form;
