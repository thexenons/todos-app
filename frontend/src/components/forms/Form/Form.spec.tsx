import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import Form from ".";

describe("Form", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<Form onSubmit={() => null} />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
