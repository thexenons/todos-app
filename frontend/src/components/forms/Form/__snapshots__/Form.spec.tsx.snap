// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form > Render 1`] = `
<div>
  <form />
</div>
`;
